driver/src/test/java/org/neo4j/driver/integration/async/AsyncTransactionIT.java

```diff
@@ -679,53 +669,53 @@ void shouldFailToCommitWhenQueriesFailAndErrorNotConsumed() throws InterruptedEx
 
         ClientException e = assertThrows( ClientException.class, () -> await( tx.commitAsync() ) );
         assertNoCircularReferences( e );
-        assertEquals( "/ by zero", e.getMessage() );
+        assertEquals( "Transaction can't be committed. It has been rolled back either because of an error or explicit termination",
+                      e.getMessage() );
     }
 
     @Test
-    void shouldPropagateRunFailureFromCommit()
+    void shouldFailToCommitWhenRunFailed()
     {
         AsyncTransaction tx = await( session.beginTransactionAsync() );
 
         tx.runAsync( "RETURN ILLEGAL" );
 
         ClientException e = assertThrows( ClientException.class, () -> await( tx.commitAsync() ) );
         assertNoCircularReferences( e );
-        assertThat( e.getMessage(), containsString( "ILLEGAL" ) );
+        assertThat( e.getMessage(), containsString( "Transaction can't be committed" ) );
```

Comment:
I wonder how that even works in a non flaky way tbh.

Reply:
As far as I am aware, this test in not known as flaky at the moment.........

The internal TX keeps track of all result cursors. For instance, `runAsync` registers them [here](https://github.com/neo4j/neo4j-java-driver/blob/4.3/driver/src/main/java/org/neo4j/driver/internal/async/UnmanagedTransaction.java#L212).
The `commitAsync` executes [after](https://github.com/neo4j/neo4j-java-driver/blob/4.3/driver/src/main/java/org/neo4j/driver/internal/async/UnmanagedTransaction.java#L183) they complete.

The `RunResponseHandler` will mark the transaction as [failed](RunResponseHandler) and the `doCommitAsync` will [fail](https://github.com/neo4j/neo4j-java-driver/blob/4.3/driver/src/main/java/org/neo4j/driver/internal/async/UnmanagedTransaction.java#L261).